control_panel_api/tests/test_services.py

```diff
@@ -0,0 +1,84 @@
+import json
+from unittest.mock import patch, call
+
+from django.test.testcases import SimpleTestCase, override_settings
+
+from control_panel_api import services
+
+
+@override_settings(ENV='test', BUCKET_REGION='eu-test-2', LOGS_BUCKET_NAME='moj-test-logs')
+class ServicesTestCase(SimpleTestCase):
+    def test_policy_document_readwrite(self):
+        document = services.get_policy_document('foo', readwrite=True)
+
+        sids = [s['Sid'] for s in document['Statement']]
+        self.assertIn('UpdateRenameAndDeleteObjects', sids)
```

Comment:
IAM policies' Sid have no effects on what level of access the policy gives.
We shouldn't test for their name, we may accidentally break the policy document and tests will still pass.

I advice we do what we did in the lambda function test and we keep a copy of the expected policy documents in the tests: https://github.com/ministryofjustice/analytics-platform-ops/blob/master/infra/terraform/modules/data_access/teams/tests/conftest.py#L22

This duplication is good because if we change the policy document for whatever reason tests will fail unless we explicitly update the test policy documents.

Reply:
Also, do we want to test this method? I wonder if this method should be private, unless we want some parts of the app to get a policy document directly (I don't think we do it at the moment).